hooks/useHttp.ts

```diff
@@ -115,10 +124,21 @@ export const useStatelessFetch = <T>(
     // e.g. callbacks when unauthorized, etc...
     const status = response.status;
     const ok = response.ok;
-    const data = ok && (await response.json());
-    const error = data?.error;
 
-    return { status, ok, data, error };
+    let data: any;
+    let dataTypeError: any;
+
+    try {
+      if (options.expectedResponseType) {
```

Comment:
options di atas nda mau di spread jd gini aja jem ?

```typescript
 const {
    method,
    endpoint,
    contentType,
    expectedResponseType = 'json'
  } = options;
```
jadi nda perlu di if else in disini ?

Reply:
done pak

Reply:
sip, sudah ok harusnya. silahkan di merge